### ISSUE TITLE
Markdown special characters are transformed even inside hyperlink URLs

### ISSUE THREAD

**Issue by jnm** [enketo/enketo-transformer#145]

**Comment by MartijnR** [comment]
I wanted to keep it lightweight (it was a client-side implementation initially) and very restricted (esp HTML). It also provided an opportunity to easily match this implementation in Collect (though we may have diverged a bit since introduction): https://github.com/getodk/collect/blob/1cd96f5c67e9548471cc205bd8da706d8af87b08/collect_app/src/main/java/org/odk/collect/android/utilities/HtmlUtils.java#L70

**Comment by eyelidlessness** [comment]
> I wanted to keep it lightweight (it was a client-side implementation initially)

Makes sense! My first thought was this is potentially performance sensitive where transforms already are. For what it’s worth, in my quick spike into performance characteristics of potential third party solutions, I don’t expect this to be an issue. Apart from a surprisingly fast outlier (which unfortunately doesn’t support our use case), most Markdown libraries tend to scale with what work they do, which…

> and very restricted (esp HTML).

…actually works in our favor in an unexpected way! The library which seems to best fit this use case—[Remark](https://github.com/gnab/remark)—is generally “slower”, but that’s primarily because it’s designed as/on top of a very general purpose AST plugin system… none of which we need. We likely don’t even need most of what it comes with built in. And restricting HTML will be mostly a noop from the same AST tooling design.

I’m already familiar with the underlying tech and should be able to put together a benchmark/stress test pretty easily to validate it, but I’m pretty sure it’ll solve the problem correctly without any performance concerns.

**Comment by MartijnR** [comment]
Great. Thankfully, we have fairly comprehensive tests in enketo-transformer as well. There may be some surprises though with a library.

Fwiw, there is also an idea to bring markdown handling back to the client. https://github.com/enketo/enketo-core/issues/701 https://github.com/enketo/enketo-core/issues/554
